debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java

```diff
@@ -912,14 +920,18 @@ public Map<String, String> configs() {
                             fail("Stopping connector after error in the application's handler method: " + handlerError.getMessage(),
                                     handlerError);
                         }
+                        if (retryError != null) {
+                            fail("Stopping connector after max retry limit exceeded: " + retryError.getMessage(),
+                                    retryError);
+                        }
                         try {
                             // First stop the task ...
                             LOGGER.info("Stopping the task and engine");
                             task.stop();
                             connectorCallback.ifPresent(DebeziumEngine.ConnectorCallback::taskStopped);
                             // Always commit offsets that were captured from the source records we actually processed ...
                             commitOffsets(offsetWriter, commitTimeout, task);
-                            if (handlerError == null) {
+                            if (handlerError == null && retryError == null) {
```

Comment:
maybe this if should be moved upper entirely like this:
if (handlerError != null)  {
...
}
else if (retryError != null) {
...
}
else {
// We stopped normally ...
    succeed("Stopping connector '" + connectorClassName + "' after completed normally.");
}